pysradb/basedb.py

```diff
@@ -24,7 +25,9 @@ def __init__(self, sqlite_file):
 
     def open(self):
         """Open sqlite connection."""
-        self.db = sqlite3.connect(self.sqlite_file)
+        # Originally sqlite3.connect(self.sqlite_file)
+        self.sqlite_file = self.sqlite_file.replace('?','')
+        self.db = sqlite3.connect('file:{}?mode=rw'.format(self.sqlite_file), uri=True)
```

Comment:
Any particular reason to have the `rw` mode and not just `r`?

Reply:
`sqlite3.connect()` has `rwc` (read-write-create) mode and it gives both read-write permission to user as well as creates the file if it is not present. I changed it to `rw` mode so that it does not create a new file if it does not exist and to keep the database permissions same as the original command. If we want to give only read permission I'll change it to `ro`, please confirm.

Reply:
We do not need the write/create permission. The file is downloaded from an online resource. If it does not exist, it should just throw an error. Probably it is simpler to just check `os.path.isfile(input_file)`?

Reply:
Thanks for the clarification. I'll change the mode to `ro`. Actually, it does throw an error in case no file is found `sqlite3.OperationalError: unable to open database file` which is caught in `_verify_srametadb()` function so there is no need for a separate check.